codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -313,6 +317,62 @@ public static void setTransferEncodingChunked(HttpMessage m, boolean chunked) {
         }
     }
 
+    /**
+     * Fetch charset from message's Content-Type header.
+     *
+     * @return the charset from message's Content-Type header or {@link java.nio.charset.StandardCharsets#ISO_8859_1}
+     * if charset is not presented or unparsable
+     */
+    public static Charset getCharset(HttpMessage message) {
+        return getCharset(message, CharsetUtil.ISO_8859_1);
+    }
+
+    /**
+     * Fetch charset from message's Content-Type header.
+     *
+     * @return the charset from message's Content-Type header or {@param defaultCharset}
+     * if charset is not presented or unparsable
+     */
+    public static Charset getCharset(HttpMessage message, Charset defaultCharset) {
+        String charsetString = getCharsetAsString(message);
+        if (charsetString != null) {
+            try {
+                return Charset.forName(charsetString);
+            } catch (UnsupportedCharsetException unsupportedException) {
+                return defaultCharset;
+            }
+        } else {
+            return defaultCharset;
+        }
+    }
+
+    /**
+     * Fetch charset string from message's Content-Type header.
+     *
+     * A lot of sites/possibly clients have charset="CHARSET", for example charset="utf-8". Or "utf8" instead of "utf-8"
+     * This is not according to standard, but this method provide an ability to catch desired mistakes manually in code
+     *
+     * @return the charset string from message's Content-Type header or {@code null} if charset is not presented
+     */
+    public static String getCharsetAsString(HttpMessage message) {
```

Comment:
Is it necessary to force the `toString()` conversion? Can we just return `CharSequence` and if the user needs it in `String` form they can do the `toString()`?